test/integration/server.h

```diff
@@ -645,6 +649,9 @@ class IntegrationTestServerImpl : public IntegrationTestServer {
                                ProcessObjectOptRef process_object,
                                Buffer::WatermarkFactorySharedPtr watermark_factory) override;
 
+  // Used to skip thread checks for multi-envoy tests, when the test server is
+  // created create() with skip_asserts = true.
+  std::unique_ptr<Thread::SkipAsserts> skip_;
```

Comment:
The reason I wasn't expecting to see this was my mental model is that you'd instantiate this in main() for an application that doesn't want this. In fact I thought Envoy Mobile was doing this.

It seems weird to do this here and plumb it around. Can we do this in Envoy Mobile test infra instead? Equiv. of test_runner or something?

Reply:
I explicitly want this test here, so folks don't commit PRs here which break Envoy mobile.

Reply:
Also I'd like to have multi-envoy tests, at which point doing checks in the main thread is insufficient and this is required :-)